File: front/InterviewProjectProject/src/services/resumeService.ts
import type { ResumeData, ResumeNavItem } from '@/types/resume'
import { http } from '@/utils/http'

/**
 * 简历数据服务（已对接后端）
 *
 * 调用 Spring Boot 的 /api/resumes 接口，页面与 store 无需改动。
 */

interface ResumeSnapshot {
  resumes: Record<string, ResumeData>
  navItems: ResumeNavItem[]
}

export async function fetchResumeSnapshot(): Promise<ResumeSnapshot> {
  return http.get<ResumeSnapshot>('/api/resumes')
}

export async function saveResume(id: string, data: ResumeData): Promise<void> {
  await http.put<void>(`/api/resumes/${encodeURIComponent(id)}`, data)
}

export async function createResume(data: ResumeData, label: string): Promise<ResumeNavItem> {
  return http.post<ResumeNavItem>('/api/resumes', { data, label })
}

export async function deleteResume(id: string): Promise<boolean> {
  return http.del<boolean>(`/api/resumes/${encodeURIComponent(id)}`)
}

/**
 * 重置为初始数据（仅本地阶段使用）。后端以数据库为准。
 */
export async function resetResumes(): Promise<void> {
  // no-op
}

File: back/InterViewBacked/src/main/java/com/wjh/interviewbacked/service/ResumeService.java
package com.wjh.interviewbacked.service;

import com.wjh.interviewbacked.dto.ResumeData;
import com.wjh.interviewbacked.dto.ResumeNavItem;
import com.wjh.interviewbacked.dto.ResumeSnapshot;

public interface ResumeService {

    /**
     * 获取整份快照：resumes（id->内容） + navItems。
     * userId 为 null（匿名/游客）→ 返回脱敏的公开模板（站长的简历抹掉私人联系方式）；
     * 已登录 → 仅返回该用户自己的简历（数据隔离）。
     */
    ResumeSnapshot getSnapshot(String userId);

    /** 新建简历，返回导航项（归属当前用户） */
    ResumeNavItem createResume(ResumeData data, String label, String userId);

    /** 保存（覆盖）某份简历，校验归属（非本人 → 403） */
    void saveResume(String id, ResumeData data, String userId);

    /** 删除，返回是否成功，校验归属（非本人 → 403） */
    boolean deleteResume(String id, String userId);
}
